src/coffee/helpers/repeater.coffee

```diff
@@ -57,7 +57,9 @@ class Repeater
 
     debug '(re)-trying task, %d remaining attempts', remainingAttempts
     if remainingAttempts is 0
-      defer.reject new Error "Failed to retry the task after #{@_options.attempts} attempts. Cause: #{lastError.stack}"
+      defer.reject
```

Comment:
The backwards compatibility problem stems from the fact that a user might expect an `Error` object from the repeater. Just rejecting `lastError` wouldn't solve this problem because `lastError` can essentially be any value that was rejected in the previous attempt, not just `Error` objects.
This is also why I think that the current way of doing it is wrong anyway. You can't expect `lastError` to be an error and hence can't expect it to have a `stack` property.